lib/asyn-thread.c

```diff
@@ -754,7 +754,11 @@ void Curl_resolver_kill(struct Curl_easy *data)
   /* If we're still resolving, we must wait for the threads to fully clean up,
      unfortunately.  Otherwise, we can simply cancel to clean up any resolver
      data. */
+#ifdef _WIN32
+  if(td && (td->complete_ev || td->thread_hnd != curl_thread_t_null)
+#else
   if(td && td->thread_hnd != curl_thread_t_null
+#endif
      && (data->set.quick_exit != 1L))
```

Comment:
Since this change fixes a crash that happens if `thread_wait_resolv()` is not called, should we then not also prevent `data->set.quick_exit` from skipping the call? That option is meant to allow a memory-leak for when the process is existing anyway, but we don't want it to generate a crash...

Reply:
[alternative PR](https://github.com/curl/curl/pull/13518) does that, plus cancels the outgoing DNS request.